### PR TITLE
EPL-2 and contribute.md

### PR DESCRIPTION
**Pull request by svenefftinge**
This PR changes the code license from EPL 1.0 to EPL 2.0 and adds a Contribute.md file.

see https://github.com/eclipse/xtext/issues/1176

### CODE REVIEW

**Comment by svenefftinge** [comment]
@cdietrick In this PR I have changed both mentions of EPL 1 in the headers. Why is that not sufficient? (I know I missed one of them in another PR)

**Comment by cdietrich** [comment]
- all right reserved is still there
- SPDX-License-Identifier missing
- changes to other files also missing (about.html)

seee https://github.com/eclipse/xtext-lib/pull/297#issuecomment-581879648

**Comment by svenefftinge** [comment]
Updated

**Comment by cdietrich** [comment]
@svenefftinge miss a change in the settings for jdt/templates.
did no double check if you changed it in lib or not (sry for that)

**Comment by cdietrich** [comment]
there seems to be a failing test too

File: org.eclipse.xtext.ide.tests/model/generated/TestLanguage.genmodel [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <genmodel:GenModel xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore"
-    xmlns:genmodel="http://www.eclipse.org/emf/2002/GenModel" copyrightText="Copyright (c) 2016, 2017 TypeFox GmbH (http://www.typefox.io) and others.&#xA;All rights reserved. This program and the accompanying materials&#xA;are made available under the terms of the Eclipse Public License v1.0&#xA;which accompanies this distribution, and is available at&#xA;http://www.eclipse.org/legal/epl-v10.html"
+    xmlns:genmodel="http://www.eclipse.org/emf/2002/GenModel" copyrightText="Copyright (c) 2016, 2017 TypeFox GmbH (http://www.typefox.io) and others.&#xA;All rights reserved. This program and the accompanying materials&#xA;are made available under the terms of the Eclipse Public License v2.0&#xA;which accompanies this distribution, and is available at&#xA;http://www.eclipse.org/legal/epl-2.0"
```
Comment (szarnekow) [inline org.eclipse.xtext.ide.tests/model/generated/TestLanguage.genmodel]: The genmodel files still contain the old header.

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/EMFGeneratorFragment2Test.xtend [hunk 2]
```diff
@@ -49,30 +50,30 @@ class EMFGeneratorFragment2Test {
...
-		are made available under the terms of the Eclipse Public License v1.0
+		are made available under the terms of the Eclipse Public License v2.0
 		which accompanies this distribution, and is available at
-		http://www.eclipse.org/legal/epl-v10.html''',
+		http://www.eclipse.org/legal/epl-2.0''',
 		'''
 		/**
 		 * Copyright (c) 2011, 2019 itemis AG (http://www.itemis.eu) and others.
 		 * All rights reserved. This program and the accompanying materials
 		 * are made available under the terms of the Eclipse Public License v1.0
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/EMFGeneratorFragment2Test.xtend]: The test expectation does no longer match

File: org.eclipse.xtext.util/.settings/org.eclipse.jdt.core.prefs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+eclipse.preferences.version=1
```
Comment (szarnekow) [inline org.eclipse.xtext.util/.settings/org.eclipse.jdt.core.prefs]: I'm not sure about the newly committed preferences.

File: org.eclipse.xtext.util/.settings/org.eclipse.jdt.ui.prefs [hunk 4]
```diff
@@ -64,7 +64,7 @@ org.eclipse.jdt.ui.keywordthis=false
...
...catchblock" name\="catchblock">// ${todo} Auto-generated catch block\n${exception_var}.printStackTrace();</template><template autoinsert\="true" context\="methodbody_context" deleted\="false" description\="Code in created function stubs" enabled\="true" id\="org.eclipse.wst.jsdt.ui.text.codetemplates.methodbody" name\="methodbody">// ${todo} Auto-generated function stub\n${body_statement}</template><template autoinsert\="true" context\="constructorbody_context" deleted\="false" description\="Code in created constructor stubs" enabled\="true" id\="org.eclipse.wst.jsdt.ui.text.codetemplates.constructorbody" name\="constructorbody">${body_statement}\n// ${todo} Auto-generated constructor stub</template><template autoinsert\="true" context\="getterbody_context" deleted\="false" description\="Code in created getters" enabled\="true" id\="org.eclipse.wst.jsdt.ui.text.codetemplates.getterbody" name\="getterbody">return ${field};</template><template autoinsert\="true" context\="setterbody_context" deleted\="false" description\="Code in created setters" enabled\="true" id\="org.eclipse.wst.jsdt.ui.text.codetemplates.setterbody" name\="setterbody">${field} \= ${param};</template></templates>
```
Comment (szarnekow) [inline org.eclipse.xtext.util/.settings/org.eclipse.jdt.ui.prefs]: The file header in the prefs is not up-to-date yet

File: org.eclipse.xtext.util/plugin.properties [hunk 5]
```diff
@@ -1,9 +1,9 @@
 ###############################################################################
```
Comment (szarnekow) [inline org.eclipse.xtext.util/plugin.properties]: The file header in the plugin.properties has an outdated license header.

File: org.eclipse.xtext/src/org/eclipse/xtext/Extensions.ext [hunk 6]
```diff
@@ -1,9 +1,9 @@
 /*******************************************************************************
 * Copyright (c) 2008 itemis AG and others.
 * All rights reserved. This program and the accompanying materials
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/Extensions.ext]: The ext file has an outdated header.

**Comment by szarnekow** [comment]
This should be good to go now, @cdietrich

**Comment by szarnekow** [comment]
But let's wait for the build to finish ;)

**Comment by svenefftinge** [comment]
@szarnekow thank you for driving this home. 🙏

**Pull request merged**